### PR TITLE
Incorporate getters and setters for points and add docs

### PR DESCRIPTION
**Pull request by rexagod**
Addresses #3 #4 and (in a bit) #2 

cc @jywarren

### CODE REVIEW

**Comment by rexagod** [comment]
Also, @jywarren I've compared this branch with the `core-experimental` one which I'll merge into main once this PR is cleared. I did it so you can easily review the diffs.

**Comment by jywarren** [comment]
That's awesome @rexagod I'm so pleased!!!

File: README.md [hunk 1]
```diff
@@ -12,6 +12,51 @@
...
+Please note that the the "global `utils` object" mentioned above is returned as a parameter to the callback function from where it can be accessed. See this example:
+
+```js
+  new Matcher('path/to/image1.png', 'path/to/image2.png',
+    async function (r) { // r here is the passed utils object
+      res = await r;
+      console.log(res.getPoints); // see output below
+    });
+```
+The output `getPoints` is in the following format:
```
Comment (jywarren) [inline README.md:29]: getPoints sounds like a function but it's a property, right? If it's not a function maybe we'd better rename it just "points"?

File: README.md [hunk 2]
```diff
@@ -12,6 +12,51 @@
...
+```
+[{"x":37,"y":261},
+ {"x":482,"y":402},
+ {"x":84,"y":331}, ...]
+```
+
+`matching` is done with the `findMatchedPoints` function, passes a `matchesArray` to the global `utils` object's `getMatchedPoints` object property with the following format:
```
Comment (jywarren) [inline README.md]: This is a little hard to follow, could you add a code example showing the function in use? Thanks!!!

**Review by jywarren — CHANGES_REQUESTED** [review]
Super easy to review, thank you so much!!! Congrats, this is really great.

File: README.md (hunk 2 above)
Comment (rexagod) [reply README.md]: Cool! I've added code links to those functions since they were kind of lengthy. But let me know if you want me to copy them into the docs. Thanks!

**Comment by rexagod** [comment]
@jywarren Up for your review. 😄

File: README.md (hunk 2 above)
Comment (jywarren) [reply README.md]: Just an example of `findMatchedPoints` being run and the parameters for it would be great.

**Comment by jywarren** [comment]
Just one request then this should be good to go!

**Comment by jywarren** [comment]
Hi, are there any updates on this? Thank you!

**Comment by rexagod** [comment]
@jywarren Have a look!

File: README.md [hunk 3]
```diff
@@ -12,6 +12,53 @@
...
+```
+
+`matching` is done with the [`findMatchedPoints`](/src/orb.core.js#L241) function, which passes a `matchesArray` to the global `utils` object's `getMatchedPoints` object property with the following format:
+```
+[{"confidence":{"c1":63,"c2":187},"x1":359,"y1":48,"x2":65,"y2":309,"population":9},
+ {"confidence":{"c1":124,"c2":169},"x1":260,"y1":333,"x2":546,"y2":295,"population":9}, ...]
+```
+It runs slower than the point `finding` step due to the added computational overhead of comparing both of the images for matches.
+
+The [`findMatchedPoints`](/src/orb.core.js#L241) function depends upon the values served back into its lexical scope by the [`findPoints`](/src/orb.core.js#) function, which in turn depends upon the `params` argument (see below) supplied by the user, and is solely responsible for the generation of the [`cornersArray`](/src/orb.core.js#233), which is used to instantiate the [`matchesArray`](/src/orb.core.js#L269). The [`findMatchesPoints`](/src/orb.core.js#L241), is [called here](/src/orb.core.js#L302) and the appropriate values are set [in the cache](/src/orb.core.js#L316).
```
Comment (jywarren) [inline README.md]: Oops maybe a typo on findmatchespoints here?

Also I really do think a full example here would be ideal in addition to the linked code. It just helps explain each parameter. Thanks!

**Review by jywarren — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (rexagod) [reply README.md]: https://github.com/publiclab/matcher-core/blob/c3a5b8a6fccefc5fe7ad521eb40e25a8d9059551/src/orb.core.js#L244

@jywarren The `findMatchedPoints` function doesn't take any values explicity, those are served to it owing to the `findPoints` method. The function, however, depends specifically on `args.browser` that's initialized all the way at the top. Do you want me to highlight this?

https://github.com/publiclab/matcher-core/blob/c3a5b8a6fccefc5fe7ad521eb40e25a8d9059551/src/orb.core.js#L14

Also, every parameter has been explained in the `codeflow` section, along with their line addresses as well.

https://github.com/publiclab/matcher-core/blob/c3a5b8a6fccefc5fe7ad521eb40e25a8d9059551/README.md#codeflow

So do you want me to highlight the fact and maybe link a few lines about the working of `findMatchedPoints` (other than the text above) or modify something else? Sorry for taking your time on this PR, just wanted to be sure we are on the same page here!

Thank you!

Comment (jywarren) [reply README.md]: > The findMatchedPoints function doesn't take any values explicity, those are served to it owing to the findPoints method.

Hm, why doesn't it accept two `points` parameters, between which it finds matches? Wouldn't that make more sense to a user? Like, it would take two points collections as inputs, and output those which are matched between? If this would take extra changes, let's do that in a follow-up PR. Can you open an issue and copy these notes in? Thanks!

> The function, however, depends specifically on args.browser that's initialized all the way at the top. Do you want me to highlight this?

Yes, that doesn't seem so clear to me so it may need explaining. 

So, `Below is a summary of each component of the orbify instance` doesn't clearly spell out that these are parameters that can be submitted in the constructor. I think it's very important to literally show verbatim how to use each of the constructors, as in this example:

image

That'll make it much easier for people to start using this library, or to use it in unexpected ways! 

Thank you, @rexagod -- much appreciated!

**Comment by rexagod** [comment]
@jywarren I did a sweep of the branches avoid confusion and am shifting this PR's mergability from `core-experimental` (deleted branch) to `main`. Will continue on #10 after that. Thanks!
